diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter1/IdentifierNaming.kt

```diff
@@ -242,15 +241,13 @@ class IdentifierNaming(configRules: List<RulesConfig>) : DiktatRule(
     @Suppress("UnsafeCallOnNullableType")
     private fun extractVariableIdentifiers(node: ASTNode): List<ASTNode> {
         val destructingDeclaration = node.getFirstChildWithType(DESTRUCTURING_DECLARATION)
-        val result = if (destructingDeclaration != null) {
-            destructingDeclaration.getAllChildrenWithType(DESTRUCTURING_DECLARATION_ENTRY)
-                .map { it.getIdentifierName()!! }
-        } else if (node.parents().count() > 1 && node.treeParent.elementType == VALUE_PARAMETER_LIST &&
-                node.treeParent.treeParent.elementType == FUNCTION_TYPE) {
-            listOfNotNull(node.getIdentifierName())
-        } else {
-            listOf(node.getIdentifierName()!!)
-        }
+        val result = destructingDeclaration?.getAllChildrenWithType(DESTRUCTURING_DECLARATION_ENTRY)?.map { it.getIdentifierName()!! }
```

Comment:
I don't think this looks better, why have you changed this?

Reply:
Because it recomends to avoid `if (destructingDeclaration != null)`

Reply:
What does? Kotlin compiler?